src/status_im/transport/message/core.cljs

```diff
@@ -91,18 +52,48 @@
       (aget obj i))
     [obj]))
 
+(fx/defn handle-chat [cofx chat]
+  ;; :unviewed-messages-count is managed by status-react, so we don't copy
+  ;; over it
+  (models.chat/ensure-chat cofx (dissoc chat :unviewed-messages-count)))
+
+(fx/defn handle-message-2 [cofx message]
+  (fx/merge cofx
+            (models.message/receive-one message)
+            (ens/verify-names-from-message message (:from message))))
+
+(fx/defn process-response [cofx response-js]
+  (let [chats (.-chats response-js)
+        raw-messages (.-rawMessages response-js)
+        messages (.-messages response-js)]
+    (cond
+      (seq chats)
+      (let [chat (.pop chats)]
+        (fx/merge cofx
+                  {:dispatch-later [{:ms 20 :dispatch [::process-2 response-js]}]}
+                  (handle-chat (-> chat (clj-bean/->clj) (data-store.chats/<-rpc)))))
+      (seq raw-messages)
+      (let [first-filter (aget raw-messages 0)
+            messages     (.-messages first-filter)
+            first-message (.pop messages)]
+       ;; Pop the empty array
+        (when (= (.-length messages) 0)
+          (.pop raw-messages))
+        (when first-message
+          (fx/merge cofx
+                    {:dispatch-later [{:ms 20 :dispatch [::process-2 response-js]}]}
+                    (handle-raw-message first-message))))
+
+      (seq messages)
+      (let [message (.pop messages)]
+        (fx/merge cofx
+                  {:dispatch-later [{:ms 20 :dispatch [::process-2 response-js]}]}
+                  (handle-message-2 (-> message (clj-bean/->clj) (data-store.messages/<-rpc))))))))
+
 (handlers/register-handler-fx
- ::process
- (fn [cofx [_ messages now-in-s]]
-   (let [[chat-id message] (first messages)
-         remaining-messages (rest messages)]
-     (if (seq remaining-messages)
-       (assoc
-        (receive-message cofx now-in-s chat-id message)
-        ;; We dispatch later to let the UI thread handle events, without this
-        ;; it will keep processing events ignoring user input.
-        :dispatch-later [{:ms 20 :dispatch [::process remaining-messages now-in-s]}])
-       (receive-message cofx now-in-s chat-id message)))))
+ ::process-2
```

Comment:
why is it process-2?

Reply:
I had `:process` before, so didn't override it, but removed it after, and forgot to rename :)